src/goRunTestCodelens.ts

```diff
@@ -0,0 +1,60 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import vscode = require('vscode');
+import { CodeLensProvider, TextDocument, CancellationToken, CodeLens, Command } from 'vscode';
+import { getTestFunctions } from './goTest';
+import { GoDocumentSymbolProvider } from './goOutline';
+
+export class GoRunTestCodeLensProvider implements CodeLensProvider {
+	public provideCodeLenses(document: TextDocument, token: CancellationToken): CodeLens[] | Thenable<CodeLens[]> {
+		if (!document.fileName.endsWith('_test.go')) {
+			return;
+		}
+
+		return Promise.all([
+			this.getCodeLensForPackage(document),
+			this.getCodeLensForFunctions(document)
+		]).then(res => {
+			return res[0].concat(res[1]);
+		});
+	}
+
+	private getCodeLensForPackage(document: TextDocument): Thenable<CodeLens[]> {
+		let documentSymbolProvider = new GoDocumentSymbolProvider();
+		return documentSymbolProvider.provideDocumentSymbols(document, null)
+				.then(symbols => symbols.find(sym => sym.kind === vscode.SymbolKind.Package && !!sym.name))
+				.then(pkg => {
+					if (pkg) {
+						const range = pkg.location.range;
+						return [
+							new CodeLens(range, {
+								title: 'run package tests',
```

Comment:
I personally prefer lowercase + shorter text to not get much attention. It should be as unobtrusive as possible so devs can focus on code an not on these actions. It's like this on [C# codelens](https://github.com/OmniSharp/omnisharp-vscode/blob/f92b9dfe378418e218aa62b305e8f438193a3f34/src/features/dotnetTest.ts#L172)

Next enhancement could be `debug test` 😄